src/data/zmod/basic.lean

```diff
@@ -818,6 +818,22 @@ begin
     rw [int.cast_coe_nat, int.cast_coe_nat, nat_cast_self, sub_zero, nat_cast_zmod_val], }
 end
 
+lemma val_min_abs_neg_of_ne_half {n : ℕ} {a : zmod n} (ha : a.val ≠ n / 2) :
+  (-a).val_min_abs = -a.val_min_abs :=
+begin
+  cases n, { refl },
+  simp only [zmod.val_min_abs_def_pos, zmod.neg_val],
+  split_ifs with h h' h'' h'' h' h'' h''; try { subst h },
+  { simp },
+  { contradiction },
+  { contradiction },
+  { simp only [zero_le', not_true] at h', contradiction },
+  { exfalso, exact not_le.mpr (lt_of_le_of_ne h'' ha) (nat.half_le_of_sub_le_half h') },
+  { rw [neg_sub, nat.cast_sub], apply le_of_lt (zmod.val_lt a) },
+  { rw [nat.cast_sub (le_of_lt (zmod.val_lt a))], apply sub_sub_cancel_left },
+  { exfalso, exact h'' (nat.le_half_of_half_lt_sub (not_le.mp h')) }
+end
+
 @[simp] lemma nat_abs_val_min_abs_neg {n : ℕ} (a : zmod n) :
```

Comment:
Is the `have` statement in this proof reusable?

Reply:
Which one?

Reply:
In the proof of `nat_abs_val_min_abs_neg` there is a have a statement that looks a lot like the new lemmas you added. Can you prove one in terms of the other?

Reply:
Well at first I thought it would be easy to go from `val_min_abs_neg_of_ne_half` to `nat_abs_val_min_abs_neg` but the problem is that for the case of `a = n / 2`, we still need to make the annoying case distinction on the parity of `n`, so in the end I'm not sure if `val_min_abs_neg_of_ne_half` helps that much

Reply:
Will investigate again tomorrow

Reply:
I might do this at a later instead of in this PR. The simplification `nat_abs_val_min_abs_neg` would need the lemma that `(n / 2) = -(n / 2)` in `zmod n` if and only if `n` is odd.

Reply:
Yeah `n` is odd the lemma is always true (even when `a.val = n / 2`), so you could assume `(ha : 2 * a.val ≠ n)` to cover both cases, and then maybe you don't need the two auxiliary lemmas. (It's also true for `n = 0` but that's easy from [zmod.val_min_abs_def_zero](https://leanprover-community.github.io/mathlib_docs/data/zmod/basic.html#zmod.val_min_abs_def_zero).)

Reply:
You're right, I'll try to change it

Reply:
Yes, works nicely this way!